src/main/java/seedu/address/model/student/Matric.java

```diff
@@ -74,7 +74,6 @@ public int hashCode() {
      * @return String representation of the matriculation number
      */
     public String toString() {
-        return isEmptyMatric(matricNumber) ? "No matriculation number" : matricNumber;
+        return isEmptyMatric(matricNumber) ? "" : matricNumber;
```

Comment:
Great refactor! Now in v1.3 we do not need such simplistic and explicit messages that a field was left empty.